### PR TITLE
Refactored findLine() Implementation

### PR DESCRIPTION
**Pull request by yashTEF**
A better implementation for the findLine() method

Signed-off-by: yashTEF <yashsoc@gmail.com>

### CODE REVIEW

**Review by datho7561 — COMMENTED** [review]
Looks pretty good. I am going to try to find some time to test if this improves performance in large files

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/ListLineTracker.java [hunk 1]
```diff
@@ -74,36 +74,18 @@ private int findLine(int offset) {
...
-					right = mid - 1;
-			} else if (offset > line.offset) {
-				if (right == mid)
-					left = right;
-				else
-					left = mid + 1;
-			} else if (offset == line.offset) {
-				left = right = mid;
+		int right = fLines.size();
+		while (right - left > 1) {
```
[the remark this answers is not in the document]
Comment (yashTEF) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/ListLineTracker.java]: Hi @datho7561 I have made the suggested changes.
About the runtime not sure about the improvement in performance, but maybe the bitwise operations and lesser operations
may speed up in large files.

[the remark this answers is not in the document]
Comment (datho7561) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/ListLineTracker.java]: Okay. The code looks good. The performance seems to be around the same as before.

**Review by datho7561 — APPROVED** [review]
@yashTEF could you please rebase onto master? Once this is rebased, I think its good to merge.

**Pull request merged**